crypto/console/console.c

```diff
@@ -136,6 +136,15 @@ int openssl_console_open(void) {
     is_a_tty = 1;
     assert(CRYPTO_STATIC_MUTEX_is_write_locked(&console_global_mutex));
 
+    // Check for test environment variable first (platform-independent)
+    const char* test_mode = getenv("OPENSSL_CONSOLE_TEST_MODE");
```

Comment:
NP: I'm not sure about the name. There might be use-cases for this outside of just our testing. I'd prefer a more general name like `AWSLC_NO_TTY_DETECT`.